src/services/events/events-router.ts

```diff
@@ -40,15 +40,21 @@ eventsRouter.post("/", async (req, res, next) => {
 eventsRouter.get("/:EVENTID", async (req, res, next) => {
     const eventId = req.params.EVENTID;
     try {
-        const event = await Database.EVENTS.findOne({ eventId: eventId });
+        const unfiltered_event = await Database.EVENTS.findOne({
+            eventId: eventId,
+        });
 
-        if (!event) {
+        if (!unfiltered_event) {
             return res
                 .status(StatusCodes.NOT_FOUND)
                 .json({ error: "DoesNotExist" });
         }
 
-        return res.status(StatusCodes.OK).json(event.toObject());
+        const filtered_event = publicEventValidator.parse(
+            unfiltered_event.toJSON()
```

Comment:
why ToJSON?

Reply:
thought i needed it but apparently not 👍